### PR TITLE
osl: 1.10.9 -> 1.11.14.2

### PR DESCRIPTION
**Pull request by gebner**
###### Motivation for this change

New upstream version.

I've also fixed the seexpr and appleseed packages (they were broken before and depend on osl).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129248` at 35dc0faa run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>blender</li>
  </ul>
</details>
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>appleseed</li>
    <li>opencolorio</li>
    <li>openimageio2</li>
    <li>osl</li>
    <li>seexpr</li>
  </ul>
</details>

File: pkgs/development/compilers/osl/default.nix [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
+  nativeBuildInputs = [
+    cmake boost_static flex bison
+    python3.pkgs.pybind11
+  ];
   buildInputs = [
-     zlib openexr openimageio llvm
+    zlib openexr openimageio2
+    llvmPackages.llvm llvmPackages.clang-unwrapped
      partio pugixml
      util-linux # needed just for hexdump
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/osl/default.nix:37]:
```suggestion
    partio pugixml
    util-linux # needed just for hexdump
```

File: pkgs/development/compilers/osl/default.nix [hunk 2]
```diff
@@ -1,34 +1,44 @@
...
-     python # CMake doesn't check this?
+     python3 # CMake doesn't check this?
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/osl/default.nix:38]:
```suggestion
    python3 # CMake doesn't check this?
```

File: pkgs/development/compilers/seexpr/default.nix [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
+
   nativeBuildInputs = [ cmake ];
-  buildInputs = [ libGLU libpng zlib qt4 pythonPackages.pyqt4 bison flex ];
+
+  buildInputs = [ libpng zlib boost bison flex ];
+
   meta = with lib; {
     description = "Embeddable expression evaluation engine from Disney Animation";
-    homepage = "https://www.disneyanimation.com/technology/seexpr.html";
+    homepage = "http://wdas.github.io/SeExpr/";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/seexpr/default.nix:26]:
```suggestion
    homepage = "https://wdas.github.io/SeExpr/";
```

File: pkgs/tools/graphics/appleseed/default.nix [hunk 4]
```diff
@@ -1,52 +1,47 @@
...
-      "-DWITH_PYTHON=ON"
-      "-DWITH_DISNEY_MATERIAL=ON"
-      "-DUSE_SSE=ON"
-      "-DUSE_SSE42=ON"
-  ];
+    "-DINSTALL_TESTS=OFF"
+    "-DWITH_BENCH=OFF" # installation fails
+    "-DUSE_STATIC_BOOST=OFF"
+  ] ++ lib.optionals stdenv.hostPlatform.isx86_64 ([ "-DUSE_SSE=ON" ] ++
+      lib.optional stdenv.hostPlatform.sse4_2Support "-DUSE_SSE42=ON");
```
Comment (SuperSandro2000) [inline pkgs/tools/graphics/appleseed/default.nix:44]:
```suggestion
  ] ++ lib.optionals stdenv.hostPlatform.isx86_64 [ "-DUSE_SSE=ON" ]
  ++ lib.optional stdenv.hostPlatform.sse4_2Support "-DUSE_SSE42=ON";
```

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15.

**Comment by onny** [comment]
[OSL v1.11.17.0](https://github.com/AcademySoftwareFoundation/OpenShadingLanguage/releases/tag/v1.11.17.0) released :) This PR looks promising, any plans to continue the work?

File: pkgs/development/compilers/osl/default.nix [hunk 5]
```diff
@@ -1,34 +1,44 @@
 { clangStdenv, lib, fetchFromGitHub, cmake, zlib, openexr,
-openimageio, llvm, boost165, flex, bison, partio, pugixml,
-util-linux, python
+openimageio2, llvmPackages, boost165, flex, bison, partio, pugixml,
+util-linux, python3
 }:
 
 let boost_static = boost165.override { enableStatic = true; };
 in clangStdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/osl/default.nix:7]: Can we move this into top-level to make sure to only use this boost across the package?

**Pull request closed without merging**